lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,393 @@
+        limit.append('bhard=%s' % bhard)
+        result['xfs_quota']['bhard'] = int(bhard / 1024)
+
+    if isoft is not None and isoft != current_isoft:
+        limit.append('isoft=%s' % isoft)
+        result['xfs_quota']['isoft'] = isoft
+
+    if ihard is not None and ihard != current_ihard:
+        limit.append('ihard=%s' % ihard)
+        result['xfs_quota']['ihard'] = ihard
+
+    if rtbsoft is not None and int(rtbsoft / 1024) != current_rtbsoft:
+        limit.append('rtbsoft=%s' % rtbsoft)
+        result['xfs_quota']['rtbsoft'] = int(rtbsoft / 1024)
+
+    if rtbhard is not None and int(rtbhard / 1024) != current_rtbhard:
+        limit.append('rtbhard=%s' % rtbhard)
+        result['xfs_quota']['rtbhard'] = int(rtbhard / 1024)
+
+    if len(limit) > 0 and not module.check_mode:
+        if name == quota_default:
+            cmd = 'limit %s -d %s' % (type_arg, ' '.join(limit))
+        else:
+            cmd = 'limit %s %s %s' % (type_arg, ' '.join(limit), name)
+
+        r = exec_quota(module, cmd, mountpoint)
+        if r['rc'] != 0:
+            result['cmd'] = cmd
+            result['retvar'] = r
+            module.fail_json(msg='Could not set limits.', **result)
+        else:
+            result['changed'] = True
+    elif len(limit) > 0 and module.check_mode:
+        result['changed'] = True
+
+    module.exit_json(**result)
+
+    return True
+
+
+def quota_report(module, mountpoint, name, quota_type, used_type):
+    soft = None
+    hard = None
+
+    if quota_type == 'project':
+        type_arg = '-p'
+    elif quota_type == 'user':
+        type_arg = '-u'
+    elif quota_type == 'group':
+        type_arg = '-g'
+
+    if used_type == 'b':
+        used_arg = '-b'
+        used_name = 'blocks'
+    elif used_type == 'i':
+        used_arg = '-i'
+        used_name = 'inodes'
+    elif used_type == 'rtb':
+        used_arg = '-r'
+        used_name = 'realtime blocks'
+
+    r = exec_quota(module, 'report %s %s' % (type_arg, used_arg), mountpoint)
+
+    if r['rc'] != 0:
+        module.fail_json(msg='Could not get quota report for %s (%s).' % (used_name, r['stderr']))
+    else:
+        for line in r['stdout']:
+            line = line.strip().split()
+            if len(line) > 3 and line[0] == name:
+                soft = int(line[2])
+                hard = int(line[3])
+                break
+
+    return soft, hard
+
+
+def exec_quota(module, cmd, mountpoint):
+    cmd = ['xfs_quota', '-x', '-c'] + [cmd, mountpoint]
+    (rc, stdout, stderr) = module.run_command(cmd, use_unsafe_shell=True)
+    if "XFS_GETQUOTA: Operation not permitted" in stderr.split('\n') or \
+            rc == 1 and 'xfs_quota: cannot set limits: Operation not permitted' in stderr.split('\n'):
+        module.fail_json(msg='You need to be root or have CAP_SYS_ADMIN capability to perform this operation')
+
+    return {'rc': rc, 'stdout': stdout.split('\n'), 'stderr': stderr.split('\n')}
```

Comment:
No need to split stdout and stderr here. We would rather return the raw data.
```suggestion
    return rc, stdout, stderr
```